code/src/beetle.cpp

```diff
@@ -31,81 +31,163 @@ namespace r2d2::moving_platform {
+        bool m0_state = false;
+        uint_fast16_t counter_m0 = 0;
+        bool m1_state = false;
+        uint_fast16_t counter_m1 = 0;
         // Encode frequency for 1 turn of the wheel. the encoder has 64 point
         // per over 2 pins we count when de adc goes from low to high of 1 pin.
         // The gear ratio from the motor is 50:1 64/4*50 = 800
-        int encode_1_full_turn = 800;
+
         // Motor speed
         // Default turn speed.
-        int motor_speed = 20;
+        uint8_t motor_speed = move_speed;
         // Turn the right way.
         if (degrees < 0) {
             // turn right
             motor_speed = -motor_speed;
             degrees = -degrees;
         }
         // motor tests strats motors.
-        qik_2s12v10_motorcontroller.set_m0_speed(motor_speed);
+        qik_2s12v10_motorcontroller.set_m0_speed(-motor_speed);
         qik_2s12v10_motorcontroller.set_m1_speed(motor_speed);
 
         while (true && degrees != 0) {
             if (motor_encoder_m0.read() > adc_voltage) {
-                if (low_m0 == true) {
+                if (m0_state == false) {
                     counter_m0++;
                 }
-                low_m0 = false;
+                m0_state = true;
             } else {
-                low_m0 = true;
+                m0_state = false;
             }
 
-            if (counter_m0 ==
-                (int(encode_1_full_turn * turn / 360 * degrees))) {
+            if (counter_m0 == (uint_fast16_t(encode_1_full_turn * turn_factor /
+                                             360 * degrees))) {
                 qik_2s12v10_motorcontroller.brake_m0(20);
             }
 
             if (motor_encoder_m1.read() > adc_voltage) {
-                if (low_m1 == true) {
+                if (m1_state == false) {
                     counter_m1++;
                 }
-                low_m1 = false;
+                m1_state = true;
             } else {
-                low_m1 = true;
+                m1_state = false;
             }
 
-            if (counter_m1 ==
-                (int(encode_1_full_turn * turn / 360 * degrees))) {
+            if (counter_m1 == (uint_fast16_t(encode_1_full_turn * turn_factor /
+                                             360 * degrees))) {
+
                 qik_2s12v10_motorcontroller.brake_m1(20);
             }
-            if (counter_m0 > (encode_1_full_turn * turn / 360 * degrees) &&
-                counter_m1 > (encode_1_full_turn * turn / 360 * degrees)) {
+
+            if (counter_m0 >
+                    (encode_1_full_turn * turn_factor / 360 * degrees) &&
+                counter_m1 >
+                    (encode_1_full_turn * turn_factor / 360 * degrees)) {
                 break;
             }
-            // wait so the while loop aint blocking
-            hwlib::wait_ms(0.1);
         }
     }
-} // namespace r2d2::moving_platform
\ No newline at end of file
+    void beetle_c::move(uint16_t distance) {
```

Comment:
there is a missing newline between two function definitions

Reply:
https://github.com/R2D2-2019/moving_platform/pull/30/commits/8070a9986ffcbc2c2bc39b0dd23e8bc5519e25c1